modules/asm/scripts/install_asm.sh

```diff
@@ -15,59 +15,29 @@
+if [ "$#" -lt 4 ]; then
     >&2 echo "Not all expected arguments set."
     exit 1
 fi
 
 PROJECT_ID=$1
 CLUSTER_NAME=$2
 CLUSTER_LOCATION=$3
-ASM_RESOURCES=$4
-ASM_VERSION=$5
-PROJECT_NUM=$6
+ASM_VERSION=$4
+MODE="install"
 BASE_DIR="asm-base-dir"
-# check for needed binaries
-# kustomize is a requirement for installing ASM and is not available via gcloud. Safely exit if not available.
-if [[ -z $(command -v kustomize) ]]; then
-  echo "kustomize is unavailable. Skipping ASM installation. Please install kustomize, add to PATH and rerun terraform apply."
-  exit 1
-fi
-# # check docker which is optionally used for validating asm yaml using gcr.io/kustomize-functions/validate-asm:v0.1.0
-# if [[ $(command -v docker) ]]; then
-#   echo "Docker is available. ASM yaml validation will be performed."
-# else
-#   echo "ASM yaml validation will be skipped as Docker is unavailable"
-#   SKIP_ASM_VALIDATION=true
-# fi
-mkdir -p "${ASM_RESOURCES}"
-pushd "${ASM_RESOURCES}"
-gcloud config set project "${PROJECT_ID}"
-if [[ -d ./asm-patch ]]; then
-    echo "ASM patch directory exists. Skipping download..."
+
+if [[ -d "${BASE_DIR}" ]]; then
+    echo "working directory exists. Skipping creating it"
 else
-    echo "Downloading ASM patch"
-    kpt pkg get https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages.git/asm-patch@"${ASM_VERSION}" .
+    echo "creating working directory ${BASE_DIR}"
+  # make working directory
+  mkdir -p "${BASE_DIR}"
 fi
-gcloud beta anthos export "${CLUSTER_NAME}" --output-directory ${BASE_DIR} --project "${PROJECT_ID}" --location "${CLUSTER_LOCATION}"
-kpt cfg set asm-patch/ base-dir ../${BASE_DIR}
-kpt cfg set asm-patch/ gcloud.core.project "${PROJECT_ID}"
-kpt cfg set asm-patch/ gcloud.container.cluster "${CLUSTER_NAME}"
-kpt cfg set asm-patch/ gcloud.compute.location "${CLUSTER_LOCATION}"
-kpt cfg set asm-patch/ gcloud.project.environProjectNumber "${PROJECT_NUM}"
-kpt cfg list-setters asm-patch/
-pushd ${BASE_DIR}
-kustomize create --autodetect --namespace "${PROJECT_ID}"
-popd
-pushd asm-patch
-kustomize build -o ../${BASE_DIR}/all.yaml
-popd
-# # skip validate as we should investigate if we can check this without having to resort to dind
-# if [[ ${SKIP_ASM_VALIDATION} ]]; then
-#  echo "Skipping ASM validation..."
-# else
-#  echo "Running ASM validation..."
-#  kpt fn source ${BASE_DIR} | kpt fn run --image gcr.io/kustomize-functions/validate-asm:v0.1.0
-# fi
-gcloud beta anthos apply ${BASE_DIR}
-kubectl wait --for=condition=available --timeout=600s deployment --all -n istio-system
+
+#download the correct version of the install_asm script
+curl https://storage.googleapis.com/csm-artifacts/asm/install_asm_"${ASM_VERSION}" > install_asm
+chmod u+x install_asm
+
+#run the script with appropriate flags
+./install_asm --verbose --project_id ${PROJECT_ID} --cluster_name ${CLUSTER_NAME} --cluster_location ${CLUSTER_LOCATION} --mode ${MODE} --output_dir ${BASE_DIR} --enable_cluster_labels --enable_cluster_roles
```

Comment:
Is an output dir necessary, IIRC the asm script can create a tmp dir to hydrate the configs?
/cc @zerobfd

Reply:
Yep, by default it creates a /tmp directory and re-downloads the tarball and kpt pkg every time.

If you specify a stable output directory and it already has the kpt package and binary, it won't re-download anything, so you could use that to pin on a certain version until someone deleted the directory.